vite.config.ts

```diff
@@ -41,6 +47,7 @@ export default defineConfig({
 			: remix({
 					ignoredRouteFiles: ['**/*'],
 					serverModuleFormat: 'esm',
+					future: { v3_singleFetch: true },
```

Comment:
Hi, I think there's a huge problem with this.

The code is:

```ts
process.env.NODE_ENV === 'test'
  ? null
  : remix({
    future: { v3_singleFetch: true },
    // ...
  });
```

So, in the test environment, we're going to test something that isn't identical to the production or development environments.

I saw the problem because in the development environment, **MSW mocks don't work with `singleFetch` enabled**. But they do in test mode.

I think this is an issue we need to resolve before enabling `singleFetch` functionality. I don't have a solution at the moment.

Reply:
> MSW mocks don't work with singleFetch enabled.

Hmmm... Can you be more descriptive? What is not working? What output do you get?

@kettanaito, have you tried a Remix app with `singleFetch` enabled with MSW?

Reply:
@kentcdodds, I don't believe I have.

I'd love to hear more info about any errors or other behaviors MSW shows when the single fetch mode is enabled. 

Afaik, single fetch should have no effect for MSW. It groups loader resource calls for nested routes, but has no effect on the requests performed within the loaders. Then, it groups all the loader resources into, well, a single fetch, and uses a "stream fetch" to send it back to the browser. That encoded, streamed response is not something you should be intercepting. Correct me if I'm wrong about that.

Reply:
Following the [Debugging runbook](https://mswjs.io/docs/runbook) can be helpful to figure out what and why doesn't work when intercepting requests. @davidesigner, could you please let me know how far you get with that runbook?

Reply:
Wow I realised why I had this problem, it's because I had added: `installGlobals({ nativeFetch: true });` (which wasn't necessary). When I remove this line from the `vite.config.ts` file, everything works fine.

Sorry for wasting your time...

Reply:
No time wasted! I'm glad to hear that it works as expected now. Great find!